### PR TITLE
TST: bare pytest raises

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] ref #30999 
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/window/test_rolling.py [hunk 1]
```diff
@@ -113,7 +125,10 @@ def test_numpy_compat(self, method):
     def test_closed(self):
         df = DataFrame({"A": [0, 1, 2, 3, 4]})
         # closed only allowed for datetimelike
-        with pytest.raises(ValueError):
+
+        msg = "losed only implemented for datetimelike and offset based windows"
```
Comment (MarcoGorelli) [inline pandas/tests/window/test_rolling.py]: `closed` instead of `losed`?

File: pandas/tests/window/test_rolling.py [hunk 2]
```diff
@@ -38,17 +38,26 @@ def test_constructor(self, which):
...
+            "passed window foo is not compatible with a datetimelike index|"
+            "min_periods must be an integer|"
+            "center must be a boolean"
+        )
         for w in [2.0, "foo", np.array([2])]:
-            with pytest.raises(ValueError):
+            with pytest.raises(ValueError, match=msg):
                 c(window=w)
-            with pytest.raises(ValueError):
+            with pytest.raises(ValueError, match=msg):
```
Comment (simonjayhawkins) [inline pandas/tests/window/test_rolling.py]: can you separate out the messages to be more explicit

File: pandas/tests/window/test_rolling.py [hunk 3]
```diff
@@ -38,17 +38,26 @@ def test_constructor(self, which):
         c(window=2, min_periods=1, center=False)
 
         # GH 13383
-        with pytest.raises(ValueError):
+
+        msg = "window must be non-negative"
+
+        with pytest.raises(ValueError, match=msg):
```
Comment (simonjayhawkins) [inline pandas/tests/window/test_rolling.py]: the c(0) below looks misplaced. It doesn't raise. can you co-locate with the #valid statements above while your touching this.

File: pandas/tests/window/test_rolling.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/tests/window/test_rolling.py]: yes, thank you :)

**Review by simonjayhawkins — APPROVED** [review]
@MomIsBestFriend lgtm

**Review by datapythonista — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @MomIsBestFriend

**Pull request merged**